src/base.rs

```diff
@@ -38,14 +38,19 @@ pub trait Type {
 
     /// Parse `GlobalID`.
     fn from_global_id(id: ID) -> Option<ID> {
-        let v: Vec<&str> = id.splitn(2, ':').collect();
-        if v.len() != 2 {
-            return None;
-        }
-        if v[0] != Self::type_name() {
-            return None;
-        }
-        Some(v[1].to_string().into())
+        base64::decode(id.as_str())
```

Comment:
This is a weird pattern to force people into. If you structure data properly, this isn't needed because you can do a trip to the database to find the type. You shouldn't be relying on types sent from the client.

Reply:
I removed `Type::global_id` and `Type::from_global_id`. 🙂